admiral/pkg/clusters/shard_handler.go

```diff
@@ -208,12 +237,103 @@ func ConsumeIdentityConfigs(ctxLogger *log.Entry, ctx context.Context, configWri
 	}
 }
 
-// UpdateShard reads the job object from the results channel and updates the original shard object with the proper result.
-func UpdateShard(ctxLogger *log.Entry, results <-chan *ConfigWriterData, resultswg *sync.WaitGroup, shard *admiralapiv1.Shard) {
+// ProcessResults reads the data object from the results channel and updates the original shard object with the proper result.
+func ProcessResults(ctx context.Context, ctxLogger *log.Entry, results <-chan *ConfigWriterData, resultswg *sync.WaitGroup, shard *admiralapiv1.Shard, cc admiralapi.Interface) error {
 	defer resultswg.Done()
-	for job := range results {
-		ctxLogger.Infof(common.CtxLogFormat, "UpdateShard", shard.Name, "", job.ClusterName, job.Result)
-		//ctxLogger.Infof(common.CtxLogFormat, "UpdateShard", shard.Name, "", job.ClusterName, shard.Status.Conditions[0].Message)
-		//TODO: need to get updated shard crd spec and set status here
+	updatedShard := shard.DeepCopy()
+	updatedShard.Status.ClustersMonitored = len(shard.Spec.Clusters)
+	updatedShardStatusCondition := admiralapiv1.ShardStatusCondition{
+		Message:         "Shard handling complete",
+		Reason:          admiralapiv1.Processed,
+		Status:          admiralapiv1.TrueConditionStatus,
+		Type:            admiralapiv1.SyncComplete,
+		LastUpdatedTime: v1.Time{Time: time.Now()},
+	}
+	updatedShard.Status.FailureDetails = admiralapiv1.FailureDetails{
+		LastUpdatedTime: v1.Time{Time: time.Now()},
+		FailedClusters:  []admiralapiv1.FailedCluster{},
+	}
+	clusterFailedIdentitiesMap := make(map[string][]admiralapiv1.FailedIdentity)
+	for data := range results {
+		if data.Error != nil {
+			ctxLogger.Infof(common.CtxLogFormat, "ProcessResults", shard.Name, common.GetOperatorSyncNamespace(), data.ClusterName, data.Error.Error())
```

Comment:
Do you think this can be a warn or error instead of info ?

Reply:
will change it to warn, we already errorf in ConsumeIdentityConfigs where the error occurs